website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml

```
@@ -60,3 +59,7 @@ Resources:
               - ec2:DescribeSpotPriceHistory
               - ssm:GetParameter
               - pricing:GetProducts
+          - Effect: Allow
+            Action:
+              - iam:PassRole
+            Resource: !Sub "arn:aws:iam::${AWS::AccountId}:role/KarpenterNodeRole-${ClusterName}"
```

Is it possible to !Ref the role and get the ARN of the output of it?

---

This doesn't work:

```
          - Effect: Allow
            Action:
              - iam:PassRole
            Resource: !Ref "KarpenterNodeRole-${ClusterName}"
```

---

Ah just trying it myself, doesn't output as ARN. Looks like the join func works here though.
```suggestion
            Resource: !Join [ "/", [ !Sub "arn:aws:iam::${AWS::AccountId}:role", !Ref "KarpenterNodeRole" ] ]
```

---

make sure to use `${AWS::Partition}` as well:

`!Sub "${AWS::Partition}:iam::${AWS::AccountId}:role/KarpenterNodeRole-${ClusterName}"`

---

If you change the name, you'll have to change it multiple times anyway, we reference that name with the `eksctl create iamidentitymapping`. I think the `!Join` is a lot less readable.

---

I'm only talking about the CFN resources, rather than the whole setup process. If I need to change the name of a resource, then I only need to do it where the name is defined with the join. I agree it's less readable, but it will reduce user errors if they need to configure this, especially since this might be hard to debug.

---

I'd agree that the readability trade-off isn't really worth it here. This is just an example infra setup, so it should bias towards readability rather than extensibility.